mmdet/apis/train.py

```diff
@@ -68,6 +68,67 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def auto_scale_lr(cfg, distributed, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): Training config.
+        distributed (bool): Using distributed or not.
+        logger (logging.Logger): Logger.
+    """
+    warning_msg = 'in your configuration file. Please update all the ' \
+                  'configuration files to mmdet >= 2.23.1. ' \
+                  'Disable automatic scaling of learning rate.'
+
+    # Get the config of auto-scale-lr
+    if 'auto_scale_lr' not in cfg:
+        logger.warning(f'Can not find "auto_scale_lr" {warning_msg}')
+        return
+
+    # Get flag from config
+    auto_scale_lr_flag = cfg.auto_scale_lr.get('enable', False)
+    if auto_scale_lr_flag is False:
+        logger.info('Automatic scaling of learning rate (LR)'
+                    ' has been disabled.')
+        return
```

Comment:
Can merge those two branches togather.
```
if ('auto_scale_lr' not in cfg) or (not cfg.auto_scale_lr.get('enable', False)):
    ....
```
This will make the logic simpler.

Reply:
Using `or` may cause an error when cfg has not `auto_scale_lr`.

Reply:
Using or may cause an error when cfg has not auto_scale_lr.

Reply:
If the first `('auto_scale_lr' not in cfg)` is True, the second part of `(not cfg.auto_scale_lr.get('enable', False))` will not call.

Reply:
Done